src/utils.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Gets the inner relevant node (CallExpression, Identity, et al.) given a generic expression node
+ * await someAsyncFunc() => someAsyncFunc()
+ * someElement as HTMLDivElement => someElement
+ *
+ * @param {Object} expression - An expression node
+ * @returns {Object} - A node
+ */
+export function getInnerNodeFrom(expression) {
+  return expression.type === "TSAsExpression"
+    ? getInnerNodeFrom(expression.expression)
+    : expression.type === "AwaitExpression"
+    ? getInnerNodeFrom(expression.argument)
+    : expression;
+}
```

Comment:
this seems different than from what was in prefer-in-document: 

```
function getAssignmentFrom(expression) {
    return expression.type === "TSAsExpression"
      ? getAssignmentFrom(expression.expression)
      : expression.type === "AwaitExpression"
      ? getAssignmentFrom(expression.argument)
      : expression.callee
      ? expression.callee
      : expression;
  }
```

is expression.callee no longer needed?

Reply:
I had to make it return a slightly broader object because I needed access to the whole `CallExpression` in prefer-to-have-values to get access to its arguments as well as the callee. I've tweaked its consumption in prefer-in-document to maintain passivity here https://github.com/testing-library/eslint-plugin-jest-dom/pull/122/commits/e77f6b9f8dc3430743ccf2a5ae60224bb101e830#diff-6eab9c599f2637bd770a8d963e5c787cfd559888fe0c7b71fbab69d5d2bfff4bR160